### PR TITLE
Add some initial tests for JSON modules.

### PR DESCRIPTION
**Pull request by Ms2ger**
See <https://github.com/whatwg/html/pull/4407>.

### CODE REVIEW

**Comment by Ms2ger** [comment]
Note to self: some CORS tests could be useful too.

CC  @littledan

**Comment by annevk** [comment]
Chrome added some logic for `application/json+...` if I remember correctly for CORB. Might be good to verify those fail.

File: html/semantics/scripting-1/the-script-element/json-module/parse-error.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>JSON modules: Content-Type</title>
```
Comment (domenic) [inline html/semantics/scripting-1/the-script-element/json-module/parse-error.html]: Wrong title

**Review by domenic — APPROVED** [review]
LGTM as a start. For posterity, additional ideas for tests (potentially best done by the first implementer):

- Non-JSON content types get rejected, including tricky cases like @annevk pointed out.
- Test that there are no non-default imports. (Maybe easiest to test with `import()`.)
- Test non-Window contexts. (Easy: using `.any.js` + `import()`. Harder: using module workers which nobody ships yet, but Chrome has behind a flag.)
- Test JSON files containing non-objects (null, true, false, array)

**Comment by domenic** [comment]
Also there are a lot of tests in the JS modules directory that might benefit from porting. For example testing that if you have module A with side effects, that imports JSON module B containing invalid JSON, asserting that the side effects of A do not happen.

/cc @nyaxt and @hiroshige-g as folks who helped with a lot of the test coverage for JS modules.

**Pull request merged**